source/tutorials/nix-language.md

```diff
@@ -0,0 +1,324 @@
+
+The path is relative to the file containing the expression.
+
+#### Absolute path
+
+    /absolute/path
+
+#### Home directory path
+
+    ~/path/in/home/directory
+
+#### Named paths
+
+Also known as “angle bracket syntax”.
+
+    <name>
+
+Examples:
+
+    # write a Nix expression to a file
+    echo 123 > nix_path_file
+
+    # set `NIX_PATH` to point a name to that file
+    env NIX_PATH=name_in_nix_path=nix_path_file nix repl
+
+    nix repl> import <name_in_nix_path>
+    123
+
+Nix simply reads and parses the `NIX_PATH` environment variable.
+Entries are separated by colons (`:`).
+
+Note: The value of a named path depends on external system state. Therefore it is recommended not to use them.
+
+#### Path concatenation
+
+    /nix + /store
+
+Paths can be concatenated with each other or with strings.
+
+Examples:
+
+    nix-repl> /nix + /store
+    /nix/store
+
+    nix-repl> /nix/var + "/nix"
+    /nix/var/nix
+
+### Let bindings
+
+    nix-repl> let a = 1; in a + a
+    2
+
+Let bindings allow declaring named values for repeated use.
+
+### Recursive sets
+
+Recursive sets allow elements to refer to attributes within the set.
+They are constructed with the `rec` keyword.
+
+    nix-repl> rec { a = 1; b = a + a; }
+    { a = 1; b = 2; }
+
+Note: This construct
+- allows infinite recursion, such as `rec { x = x; }.x`
+- can quickly become hard to understand.
+Therefore it is recommended to use let bindings instead.
+
+    nix-repl> let x = 1; in { a = x; b = x + x; }
+    { a = 1; b = 2; }
+
+## Functions
+
+The Nix language is Turing-complete.
+It gives you the power to tame complexity and the freedom to create overwhelming amounts of it.
+
+Therefore, it is recommended to use functions sparingly and instead think more in terms of data, because data is usually easier to understand.
+
+
+### Arguments
+
+Nix functions take exactly one argument.
+
+    x: x + 1
+
+Argument and function body are separated by a colon `:`.
+
+Applying a function to an argument means writing the argument after the function name.
+
+Example:
+
+    nix-repl> let
+              f = x: x + 1;
+              in
+              f 1
+    2
+
+Arguments can be chained.
+
+    x: y: x + y
+
+The above function takes one argument and returns a function `y: x + y` with `x` set to passed value. 
+
+Example:
+
+    nix-repl> let
+              f = x: y: x + y;
+              in
+              f 1
+    «lambda @ (string):2:8»
+
+The `lambda` indicates the resulting value is a function.
+
+Applying that to another argument yields the inner body `x + y`, which can now be fully evaluated.
+
+    nix-repl> let
+              f = x: y: x + y;
+              in
+              f 1 2
+    3
+
+### Keyword arguments
+
+Nix functions can explicitly take an attribute set as argument.
+
+    {a, b}: a + b
+
+This is equivalent to
+
+    x: x.a + a.b
+
+The argument defines the exact attributes that have to be in that set.
+Leaving out or passing additional attributes is an error.
```

Comment:
```suggestion
While this is similar to

    x: x.a + x.b

only keyword arguments define the exact attributes that have to be in that set.
Leaving out or passing additional attributes is an error.
```

Bu we may as well leave out the similarity, as it introduces more detail than strictly necessary. I thought it may help readers transition from what may be structure of the argument in a function declaration.